yetanotherco/aligned_layer
# feat(sdk): Add Max Fee Estimate. Update zkQuiz to better represent how the SDK should be used

Reopens #1045 due to merge issues.

## To Test
- Run test in batcher `cd batcher && cargo t`

## To test zkQuiz example in Devnet
* Start anvil, aggregator, operator and batcher as usual
* Go to the zkquiz examples folder
```bash
cd examples/zkquiz
```
* Change the `.env` file located in `examples/zkquiz/contracts`:
```bash
RPC_URL=http://localhost:8545
PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
ALIGNED_SERVICE_MANAGER_ADDRESS=0x1613beB3B2C4f22Ee086B2b38C1476A3cE7f78E8
BATCHER_PAYMENT_SERVICE_ADDRESS=0x7969c5eD335650692Bc04293B07F5BF2e7A673C0
```
* Deploy the verifier contract. Run it in the `zkquiz` folder. Note that this will output the deployed contract address. Make sure to save it
```bash
make deploy_verifier
```
* Change the `Makefile` environment variables to the following:
```bash
CONTRACT_ADDRESS=<YOUR_DEPLOYED_CONTRACT_ADDRESS>
RPC_URL=http://localhost:8545
KEYSTORE_PATH=../../../../config-files/devnet/keys/operator-3.ecdsa.key.json
```
* Now change some code in `examples/zkquiz/quiz/script/src/main.rs`:

Line 21:
```rust
const BATCHER_URL: &str = "ws://localhost:8080";
```

Line 24:
```rust
const NETWORK: Network = Network::Devnet;
```

Line 34:
```rust
        default_value = "http://localhost:8545"
```

Line 53:
```rust
.with_chain_id(31337u64);
```

* Now send proofs so that when the zkquiz proof is submitted, it is included in a batch and posted
```bash
make batcher_send_burst_groth16
```

* Finally, run zkquiz
```bash
make answer_quiz
```

**HELP**: Answers are C C A
